Make pop up window with collaboration

mohamedgamaleldin999999 opened the pull request:
in this PR, I am trying to add detail pop up window for both mobile and desktop versions

LionRouge1 commented on styles.css:
```diff
@@ -302,13 +304,146 @@ h3 {
+  z-index: 2000;
+}
+
+.model {
+  display: none;
+  position: absolute;
+  top: 50%;
+  left: 50%;
+  transform: translate(-50%, -50%);
+  background-color: #fff;
+  z-index: 3000;
+}
```
- [ ] Nice work designing the module popup :+1: However your design is a little different from the Figma design:
  - Note the background of the module popup overlay it's `background: #C1C7D0;` on Figma design but yours is dark
  - Note the position of the module popup buttons on the desktop version, They are align horizontally with the project name but yours are at the bottom.
  - The height of the image on your module popup design on desktop version, it is a little bigger than the Figma design own
 ### Your design :point_down: 
![Screenshot from 2023-05-25 08-19-53](https://github.com/mohamedgamaleldin999999/mohamedgamaleldin999999.github.io/assets/91142432/ed483ef7-31a1-468e-aee5-21db4fd3c0aa)
### What it is supposed to look like :point_down: 
![Screenshot from 2023-05-25 09-32-55](https://github.com/mohamedgamaleldin999999/mohamedgamaleldin999999.github.io/assets/91142432/182699d2-23fe-439d-ace1-8fa524fd3f6e)

- [ ] Kindly make sure your module popup doesn't scroll with the page by adding `position: fixed` and `overflow-y: scroll` properties. It make your web site look messy
_**HIT: Your code could look like this:**_
```css
.model {
    display: none;
    position: fixed;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
    background-color: #fff;
    z-index: 3000;
    height: 90vh;
    overflow-y: scroll;
}
```

LionRouge1 reviewed (changes requested):
Hi Team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

![almost](https://media.giphy.com/media/KyBesf8kNT6s88SEgb/giphy.gif)

## Highlights
- Good PR
- Good commit
- All linters passed

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

mohamedgamaleldin999999 commented on index.html:
```diff
@@ -86,145 +86,91 @@ <h1>Hey there. I'm Jimmy.</h1>
+                    <ul>
+                      <li>Codekit</li>
+                      <li>GitHub</li>
+                      <li>JavaScript</li>
+                      <li>Bootstrap</li>
+                      <li>Terminal</li>
+                      <li>Codepen</li>
+                    </ul>
+                  </div>
+                  <p>
+                    Lorem Ipsum is simply dummy text of the printing and typesetting
+                    industry. Lorem Ipsum has been the industry's standard dummy
+                    text ever since the 1500s, when an unknown printer took a galley
+                    of type and scrambled it 1960s.
+                  </p>
+                  <p>
+                    Lorem Ipsum is simply dummy text of the printing and typesetting
+                    industry. Lorem Ipsum has been the industry's standard dummy
+                    text ever since the 1500s, when an unknown printer took a galley
+                    of type and scrambled it 1960s.
+                  </p>
+                </div>
+              </div>
+              <div class="model-footer desktop-flex">
+                <button class="btn-live">
+                  See live
+                  <span
+                    ><i
+                      class="fa-thin fa-arrow-up-left-from-circle fa-rotate-90"
+                    ></i
+                  ></span>
+                </button>
+                <button class="btn-source">
+                  See source <span><i class="fa-brands fa-github"></i></span>
+                </button>
+              </div>
+            </div>
+          </div>
+          <div id="overlay"></div>
+        </section>
```
Thank you so much, your comments are really helpful. But can I ask you to elaborate more on this, please? I feel  a little bit lost

LionRouge1 replied:
Hi, @mohamedgamaleldin999999, Sorry for the delay.

mohamedgamaleldin999999 replied:
@LionRouge1 no don't worry about it. Please let me know once you can.

LionRouge1 replied:
Right now you are showing the same information on the module popup windows for all the projects, but it supposes to have the information about the project that the user clicks. 
Ex: As a user, I clicked on the `See Project` button of Project 3, I suppose to get a popup containing information about that particular project (project 3).

There are many ways to achieve this depending on your level and choices. Therefore providing a HIT may confuse you. And I know that you can do it :smiley_cat: :sunglasses:  

If you still having problems implementing this or not still understanding what it's supposed to be done, kindly reach out to me on Slack. 

Good luck :partying_face:

mohamedgamaleldin999999 commented:
Hello my friend. Thanks for your wonderful code review. I had an inquiry
about your first comment and I commented. Can you please read and help me.
Thank you.

On Thu, 25 May 2023, 12:56 pm AVLESSI Matchoudi, ***@***.***>
wrote:

> ***@***.**** requested changes on this pull request.
>
> Hi Team,
>
> Good job so far!
> There are some issues that you still need to work on to go to the next
> project but you are almost there!
>
> [image: almost]
> <https://camo.githubusercontent.com/4f9601b422a77702aefe8e13485d2d5618cc0c30772eb2aa9f4129eac4757e06/68747470733a2f2f6d656469612e67697068792e636f6d2f6d656469612f4b7942657366386b4e5436733838534567622f67697068792e676966>
> Highlights
>
>    - Good PR
>    - Good commit
>    - All linters passed
>
> Required Changes ♻️
>
> *Check the comments under the review.*
> Optional suggestions
>
> *Every comment with the [OPTIONAL] prefix is not crucial enough to stop
> the approval of this PR. However, I strongly recommend you to take them
> into account as they can make your code better.*
>
> Cheers and Happy coding!👏👏👏
>
> Feel free to leave any questions or comments in the PR thread if something
> is not 100% clear.
> Please, remember to tag me in your question so I can receive the
> notification.
>
> *Please, do not open a new Pull Request for re-reviews. You should use the
> same Pull Request submitted for the first review, either valid or invalid
> unless it is requested otherwise.*
> ------------------------------
>
> *As described in the Code reviews limits policy
> <https://microverse.zendesk.com/hc/en-us/articles/1500004088561> you have a
> limited number of reviews per project (check the exact number in your
> Dashboard). If you think that the code review was not fair, you can request
> a second opinion using this form <https://airtable.com/shrQAqnBwek5a0O0s>.*
> ------------------------------
>
> In index.html
> <https://github.com/mohamedgamaleldin999999/mohamedgamaleldin999999.github.io/pull/14#discussion_r1205282205>
> :
>
> > +        <section id="model-section" class="model-section">
> +          <div class="model">
> +            <div class="model-content">
> +              <div class="model-header">
> +                <button class="model-close" type="button">X</button>
> +                <img
> +                  class="mobile-img"
> +                  src="./images/Snapshoot Portfolio (1).png"
> +                  alt="Model-image"
> +                />
> +                <img
> +                  class="desktop-img"
> +                  src="./images/desk-porto.png"
> +                  alt=""
> +                />
> +              </div>
> +              <div class="model-container">
> +                <div class="model-body">
> +                  <h2 class="model-text">
> +                    Keeping track of hundreds of components
> +                  </h2>
> +                  <div class="mobile-list">
> +                    <ul>
> +                      <li>Ruby on Rails</li>
> +                      <li>CSS</li>
> +                      <li>JavaScript</li>
> +                    </ul>
> +                  </div>
> +
> +                  <div class="desktop-list">
> +                    <ul>
> +                      <li>Codekit</li>
> +                      <li>GitHub</li>
> +                      <li>JavaScript</li>
> +                      <li>Bootstrap</li>
> +                      <li>Terminal</li>
> +                      <li>Codepen</li>
> +                    </ul>
> +                  </div>
> +                  <p>
> +                    Lorem Ipsum is simply dummy text of the printing and typesetting
> +                    industry. Lorem Ipsum has been the industry's standard dummy
> +                    text ever since the 1500s, when an unknown printer took a galley
> +                    of type and scrambled it 1960s.
> +                  </p>
> +                  <p>
> +                    Lorem Ipsum is simply dummy text of the printing and typesetting
> +                    industry. Lorem Ipsum has been the industry's standard dummy
> +                    text ever since the 1500s, when an unknown printer took a galley
> +                    of type and scrambled it 1960s.
> +                  </p>
> +                </div>
> +              </div>
> +              <div class="model-footer desktop-flex">
> +                <button class="btn-live">
> +                  See live
> +                  <span
> +                    ><i
> +                      class="fa-thin fa-arrow-up-left-from-circle fa-rotate-90"
> +                    ></i
> +                  ></span>
> +                </button>
> +                <button class="btn-source">
> +                  See source <span><i class="fa-brands fa-github"></i></span>
> +                </button>
> +              </div>
> +            </div>
> +          </div>
> +          <div id="overlay"></div>
> +        </section>
>
>
>    - Nice work implementing the popup window 👏 However it would be nice
>    to populate the popup windows with the data from the projects array. Kindly
>    make sure the popup windows is populate with the information of the project
>    that has been clicked.
>
> ------------------------------
>
> In script.js
> <https://github.com/mohamedgamaleldin999999/mohamedgamaleldin999999.github.io/pull/14#discussion_r1205283074>
> :
>
> > +  const projects = [
> +    {
> +      name: 'Project 1',
> +      description: 'Project 1 description',
> +      image: 'path/to/image1.jpg',
> +      technologies: ['Ruby on Rails', 'CSS', 'JavaScript', 'HTML'],
> +      liveLink: 'https://example.com/live1',
> +      sourceLink: 'https://example.com/source1',
> +    },
> +
> +    {
> +      name: 'Project 2',
> +      description: 'Project 2 description',
> +      image: 'path/to/image2.jpg',
> +      technologies: ['Ruby on Rails', 'CSS', 'JavaScript', 'HTML'],
> +      liveLink: 'https://example.com/live2',
> +      sourceLink: 'https://example.com/source2',
> +    },
> +
> +    {
> +      name: 'Project 3',
> +      description: 'Project 2 description',
> +      image: 'path/to/image2.jpg',
> +      technologies: ['Ruby on Rails', 'CSS', 'JavaScript', 'HTML'],
> +      liveLink: 'https://example.com/live2',
> +      sourceLink: 'https://example.com/source2',
> +    },
> +
> +    {
> +      name: 'Project 4',
> +      description: 'Project 2 description',
> +      image: 'path/to/image2.jpg',
> +      technologies: ['Ruby on Rails', 'CSS', 'JavaScript', 'HTML'],
> +      liveLink: 'https://example.com/live2',
> +      sourceLink: 'https://example.com/source2',
> +    },
> +
> +    {
> +      name: 'Project 5',
> +      description: 'Project 2 description',
> +      image: 'path/to/image2.jpg',
> +      technologies: ['Ruby on Rails', 'CSS', 'JavaScript', 'HTML'],
> +      liveLink: 'https://example.com/live2',
> +      sourceLink: 'https://example.com/source2',
> +    },
> +
> +    {
> +      name: 'Project 6',
> +      description: 'Project 2 description',
> +      image: 'path/to/image2.jpg',
> +      technologies: ['Ruby on Rails', 'CSS', 'JavaScript', 'HTML'],
> +      liveLink: 'https://example.com/live2',
> +      sourceLink: 'https://example.com/source2',
> +    },
> +  ];
> +
> +  const projectsContainer = document.querySelector('.cards');
> +
> +  projects.forEach((project) => {
> +    const projectElement = document.createElement('div');
> +    projectElement.classList.add('card');
> +
> +    const imageElement = document.createElement('div');
> +    imageElement.classList.add('card-img');
> +    projectElement.appendChild(imageElement);
> +
> +    const contentElement = document.createElement('div');
> +    contentElement.classList.add('info');
> +    projectElement.appendChild(contentElement);
> +
> +    const titleElement = document.createElement('h3');
> +    titleElement.classList.add('project-title');
> +    titleElement.innerHTML = "Most-Post Storie's <br/> Gain+Glory";
> +    contentElement.appendChild(titleElement);
> +
> +    const langListElement = document.createElement('ul');
> +    langListElement.classList.add('languages');
> +    contentElement.appendChild(langListElement);
> +
> +    project.technologies.forEach((technology) => {
> +      const langItemElement = document.createElement('li');
> +      langItemElement.classList.add('language');
> +      langItemElement.textContent = technology;
> +      langListElement.appendChild(langItemElement);
> +    });
> +
> +    //  'See Project' button
> +    const buttonElement = document.createElement('button');
> +    buttonElement.classList.add('view-btn');
> +    buttonElement.textContent = 'See Project';
> +    contentElement.appendChild(buttonElement);
> +
> +    // Append for projects container
> +    projectsContainer.appendChild(projectElement);
> +  });
>
>
>    - Nice work displaying the projects from the projects array 👏 However
>    it would be nice to use the data from the array to display the title for
>    each project. You can update the project name in the projects array to
>    include the title of the project or add the title property to each
>    project. And update the line 87 to display the project name or title from
>    the array.
>    *the line 87 could look like this:*
>    titleElement.innerHTML = project.name;
>
> ------------------------------
>
> In styles.css
> <https://github.com/mohamedgamaleldin999999/mohamedgamaleldin999999.github.io/pull/14#discussion_r1205283212>
> :
>
> > +.model {
> +  display: none;
> +  position: absolute;
> +  top: 50%;
> +  left: 50%;
> +  transform: translate(-50%, -50%);
> +  background-color: #fff;
> +  z-index: 3000;
> +}
>
>
>    - Nice work designing the module popup 👍 However your design is a
>    little different from the Figma design:
>       - Note the background of the module popup overlay it's background:
>       #C1C7D0; on Figma design but yours is dark
>       - Note the position of the module popup buttons on the desktop
>       version, They are align horizontally with the project name but yours are at
>       the bottom.
>       - The height of the image on your module popup design on desktop
>       version, it is a little bigger than the Figma design own
>
> Your design 👇
>
> [image: Screenshot from 2023-05-25 08-19-53]
> <https://user-images.githubusercontent.com/91142432/240869090-ed483ef7-31a1-468e-aee5-21db4fd3c0aa.png>
> What it is supposed to look like 👇
>
> [image: Screenshot from 2023-05-25 09-32-55]
> <https://user-images.githubusercontent.com/91142432/240869839-182699d2-23fe-439d-ace1-8fa524fd3f6e.png>
>
>    - Kindly make sure your module popup doesn't scroll with the page by
>    adding position: fixed and overflow-y: scroll properties. It make your
>    web site look messy
>    *HIT: Your code could look like this:*
>
> .model {
>     display: none;
>     position: fixed;
>     top: 50%;
>     left: 50%;
>     transform: translate(-50%, -50%);
>     background-color: #fff;
>     z-index: 3000;
>     height: 90vh;
>     overflow-y: scroll;
> }
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/mohamedgamaleldin999999/mohamedgamaleldin999999.github.io/pull/14#pullrequestreview-1443528189>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A5MVTKRL42UZWQ3SVR7FWOTXH4UGVANCNFSM6AAAAAAYOJQ33E>
> .
> You are receiving this because you authored the thread.Message ID:
> <mohamedgamaleldin999999/mohamedgamaleldin999999.github.
> ***@***.***>
>

Johnadibe commented on script.js:
```diff
@@ -11,8 +11,180 @@ const initApp = () => {
+          <div class='model-body'>
+            <h2>${project.name}</h2>
+            <div class='desktop-list'>
+              <ul id='desktop-componets'>
+              ${list}
+              </ul>
```
> [OPTIONAL] As this wasn't pointed out by the previous code reviewer, I noticed that on line 126, the rendered technologies list doesn't show on the popup window, I kindly request that you consider implementing that to stick with the Figma design. This ensures consistency between the design and the implementation, providing users with an intuitive and seamless experience.👍
> 
> Figma design:
> 
> ![art1](https://github.com/mohamedgamaleldin999999/mohamedgamaleldin999999.github.io/assets/40009354/384d130a-0c56-410d-9a0e-43a7f141ec84)
> 
> Your design:
> 
> ![art](https://github.com/mohamedgamaleldin999999/mohamedgamaleldin999999.github.io/assets/40009354/eab71cd7-fd3a-4bae-a81d-7707f9092288)
>

Johnadibe reviewed (approved):
# Status: Approved ✔️
### Hi @mohamedgamaleldin999999 and @Umuhan👋🏽, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

![LeonardoInGatsby](https://user-images.githubusercontent.com/38088818/233992044-ad00b20f-3f2f-4c86-8b94-1e613432af5f.gif)

## Highlights
- No linter errors✔️
- Good commit messages✔️
- Professional README Documentation✔️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
